src/main/resources/db/migration/all/20240618100000__create_cas_2_summary_views_change_id_type.sql

```diff
@@ -0,0 +1,48 @@
+-- remove id cast to TEXT
+DROP VIEW IF EXISTS cas_2_application_summary CASCADE;
+CREATE OR REPLACE VIEW cas_2_application_summary AS SELECT
+    a.id,
+    a.crn,
+    a.noms_number,
+    CAST(a.created_by_user_id AS TEXT),
+    nu.name,
+    a.created_at,
+    a.submitted_at,
+    a.hdc_eligibility_date,
+    asu.label,
+    CAST(asu.status_id AS TEXT),
+    a.referring_prison_code,
+    a.conditional_release_date,
+    asu.created_at AS status_created_at,
+    a.abandoned_at
+FROM cas_2_applications a
+LEFT JOIN (SELECT DISTINCT ON (application_id) su.application_id, su.label, su.status_id, su.created_at
+    FROM cas_2_status_updates su
+    ORDER BY su.application_id, su.created_at DESC) as asu
+    ON a.id = asu.application_id
+JOIN nomis_users nu ON nu.id = a.created_by_user_id;
+
+-- filter applications that do not have abandoned_at
+CREATE OR REPLACE VIEW cas_2_application_live_summary AS SELECT
```

Comment:
does this view need updating? I've looked at the current version and I can't see any cast of UUIDs? I suspect i'm misunderstanding something about views.

Reply:
This view `cas_2_application_live_summary` is filtered data from the first view `cas_2_application_summary`.

cas_2_application_summary.id was previously TEXT (see [here](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/main/src/main/resources/db/migration/all/20240528100000__create_cas_2_summary_views_add_abandoned.sql)).

But this cast is removed in this PR so the query will return id as UUID as per the table column type.

This is therefore consistent with the UUID type in [Cas2ApplicationSummaryEntity](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/9751be6832dfa0823ba70af677c0b6662eb912a9/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2ApplicationSummaryEntity.kt#L34).

Reply:
Hopefully this makes sense @davidatkinsuk?

Reply:
Yep all good 👍